Sources/Controllers/TargetMenu/GPX/TrackMenu/OATrackMenuHeaderView.mm

```diff
@@ -410,6 +421,24 @@ - (void)setDirection:(NSString *)direction
     self.locationSeparatorView.hidden = !hasDirection;
 }
 
+- (void)setTitle:(NSString *)title
+{
+    [self.titleView setText:title];
+
+    CGFloat titleHeight = [self.titleView.text boundingRectWithSize:CGSizeMake(self.titleView.frame.size.width, kTitleHeightMax)
+                                                            options:NSStringDrawingUsesLineFragmentOrigin
+                                                         attributes:@{ NSFontAttributeName : self.titleView.font }
+                                                            context:nil].size.height;
+    _titleHeight = titleHeight > kTitleHeightMin ? kTitleHeightMax : kTitleHeightMin;
+
+    CGRect frame = self.titleContainerView.frame;
+    frame.size.height = _titleHeight;
+    self.titleContainerView.frame = frame;
+    frame = self.titleView.frame;
+    frame.size.height = _titleHeight;
+    self.titleView.frame = frame;
```

Comment:
Two questions:
1. Is titleView without constraints? If so, why?
2. Setting a frame without animation will cause a visible "jump" on the UI. Such changes should be animated.